Document AI Obs End-user Feedback Feature (#37342)
DataDog/documentation: content/en/llm_observability/evaluations/end_user_feedback.md

---
title: End-User Feedback
description: Submit end-user feedback to Agent Observability and connect it to spans, traces, sessions, or external entities.
further_reading:
    - link: '/llm_observability/instrumentation/api/#evaluations-api'
      tag: 'Documentation'
      text: 'Learn about the Evaluations API'
    - link: '/llm_observability/evaluations/external_evaluations'
      tag: 'Documentation'
      text: 'Learn about submitting external evaluations'
    - link: '/llm_observability/evaluations/annotation_queues'
      tag: 'Documentation'
      text: 'Learn about Annotation Queues'
---

## Overview

> End-user feedback lets you bring subjective signals from the people or agents using your LLM application into Agent Observability. Examples include thumbs-up or thumbs-down ratings, whether a user accepted an agent's change, and free-text comments about a response.

Author: ```suggestion
End-user feedback lets you bring signals from the users of your LLM application into Agent Observability. Examples include thumbs-up or thumbs-down ratings, whether a user accepted an agent's change, and free-text comments about a response.
```

> Feedback is different from an evaluation. Use feedback for signals submitted by an end user or agent submitter. Use [external evaluations][1] for results produced by your own evaluator logic. Use [Annotation Queues][2] for structured review workflows run by your team.

Author: ```suggestion
Feedback is different from an evaluation. Use feedback for signals submitted by an end user. Use [external evaluations][1] for results produced by your own evaluator logic, where who submitted the evaluation is not relevant. Use [Annotation Queues][2] for structured review workflows run by your team.
```

> Submitted feedback appears in the trace side panel under the **Feedback** tab.

Author: ```suggestion
Submitted feedback will appear when viewing AI Observability sessions, traces, or spans.
```

### Submit thumbs-down feedback for a span

{{< code-block lang="json" >}}
{
  "data": {
    "type": "evaluation_metric",
    "attributes": {
      "metrics": [
        {
          "event_kind": "feedback",
          "span_id": "20245611112024561111",
          "ml_app": "weather-bot",
          "timestamp_ms": 1765990800016,
          "metric_type": "categorical",
          "label": "thumbs",
          "categorical_value": "down",
          "submitter": {
            "id": "user-123",
            "type": "user"
          }
        }
      ]
    }
  }
}
{{< /code-block >}}

### Submit free-text feedback with a feedback join key

{{< code-block lang="json" >}}
{
  "data": {
    "type": "evaluation_metric",
    "attributes": {
      "metrics": [
        {
          "event_kind": "feedback",
          "feedback_join_key": "incident-123",
          "ml_app": "incident-agent",
          "timestamp_ms": 1765990800016,
          "metric_type": "text",
          "label": "user_comment",
          "text_value": "The investigation missed the customer impact.",
          "submitter": {
            "id": "user-123",
            "type": "user"
          }
        }
      ]
    }
  }
}

> {{< /code-block >}}

Author: include assessments in these examples

## Analyze feedback

Feedback events can be queried with `@event_kind:feedback`.

> 

Author: ```suggestion
```

To create a dashboard widget for feedback, create the widget as you would for an evaluation and add the filter `@event_kind:feedback`.

> Filtering spans, traces, or sessions by feedback is not yet supported. For example, you cannot yet filter traces to only traces with thumbs-down feedback. Use dashboards and event queries scoped to `@event_kind:feedback` until feedback filtering is available.

Author: ```suggestion
Support for filtering spans, traces, or sessions by feedback will be added soon. For example, you cannot yet filter traces to only traces with thumbs-down feedback. Use dashboards scoped to `@event_kind:feedback` until feedback filtering is available.
```

Make this a note as well.
